File: frontend/src/infrastructure/ai/prompts/insightPrompt.ts
import type { AIClient } from '../provider';

const INSIGHT_SYSTEM_PROMPT = `You are an analytics assistant for a plastic recycling platform.
Given batch data, provide a concise narrative summary with actionable insights.
Return JSON: { "summary": "...", "recommendations": ["..."] }`;

export async function generateInsight(
  ai: AIClient,
  batchData: string,
): Promise<{ summary: string; recommendations: string[] }> {
  const raw = await ai.complete(batchData, INSIGHT_SYSTEM_PROMPT);
  try {
    return JSON.parse(raw);
  } catch {
    return { summary: raw, recommendations: [] };
  }
}

File: backend/aperio-api/src/infrastructure/adapters/ai/prompts/insight_prompt.py
INSIGHT_SYSTEM_PROMPT = """You are a supply chain analyst for Aperio, a plastic recycling traceability system.

Given batch data, generate a concise narrative insight that includes:
1. A summary of the batch journey through processing stages
2. Key observations about material loss at each stage
3. Any anomalies or concerns (e.g., unusually high loss)
4. Recommendations for improvement

Keep the narrative under 200 words. Be specific with numbers and percentages.
Use a professional but accessible tone.

Respond with valid JSON:
{
  "title": "Brief insight title",
  "narrative": "The full narrative text",
  "category": "batch_analysis|trend|anomaly|recommendation"
}
"""
